### PR TITLE
SDIT-994 Add create and get non-association mapping endpoints

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/NonAssociationsMappingResource.kt [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  )
+  suspend fun getNonAssociationMappingGivenNonAssociationId(
+    @Schema(description = "Non-association id", example = "2", required = true)
+    @PathVariable
+    nonAssociationId: Long,
+  ): NonAssociationMappingDto = mappingService.getNonAssociationMappingByNonAssociationId(nonAssociationId)
+
+/*
+  @PreAuthorize("hasRole('ROLE_NOMIS_NON_ASSOCIATIONS')")
+  @GetMapping("/mapping/non-associations/migrated/latest")
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/NonAssociationsMappingResource.kt:153]: Lines below commented out - will be in separate PR

**Review by steverendell — APPROVED** [review]
lgtm
